src/javascript/events/component-view.js

```diff
@@ -0,0 +1,67 @@
+/*global module, require */
+'use strict'; // eslint-disable-line strict
+
+const Core = require('../core');
+const getTrace = require('../../libs/get-trace');
+const utils = require('../utils');
+
+const decorateEventData = (eventData, viewedEl, opts) => {
+
+	const { trace, customContext } = getTrace(viewedEl);
+	let context = typeof opts.getContextData === 'function' ? opts.getContextData(viewedEl) : {};
+
+	if (typeof context !== 'object') {
```

Comment:
Is this block needed?

I guess this is guarding against consumers providing an invalid callback function, i.e. one that doesn't return an object. If that is the case then I think this could probably make people think they are using this method correctly, when in fact it should be erroring...

Reply:
I see. I will keep the block because if the function returns String, it won't throw an error, just will mess up the data detail. So I will use the block to throw an error.

Reply:
Sounds sensible!